flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -46,6 +55,20 @@
     public static final InputTypeStrategy CURRENT_WATERMARK =
             new CurrentWatermarkInputTypeStrategy();
 
+    /**
+     * Input strategy for {@link BuiltInFunctionDefinitions#JSON_OBJECT}.
+     *
+     * <p>The first argument defines the on-null behavior and is followed by any number of key-value
+     * pairs. Keys must be character string literals, while values are arbitrary expressions.
+     */
+    public static final InputTypeStrategy JSON_OBJECT =
+            compositeSequence()
+                    .argument(symbol(JsonOnNull.class))
+                    .finishWithVarying(
+                            repeatingSequence(
```

Comment:
since we have this `repeatingSequence` now, we could remove the specific one for  the `map()` function?

Reply:
I don't think that's possible because the strategy for `MAP` infers the common type over all arguments. If we used the repeating sequence strategy here, that would only happen on the first two arguments. It also has a minimum number of arguments (though this would be easy to add).